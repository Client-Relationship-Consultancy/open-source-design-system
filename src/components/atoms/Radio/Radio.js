import React from "react"
import styled from "styled-components"
import PropTypes from "prop-types"

import { colourPalette } from "brandColours"

// Declare StyledRadio as an input with type=radio with CSS
const StyledRadio = styled.input.attrs({
  type: "radio",
})`
  appearance: none;
  box-shadow: none;
  border: 2px solid ${props => props.theme.secondary.main.hex};
  outline: none;
  width: 1.5rem;
  height: 1.5rem;
  border-radius: 50%;
  transition: 0.3s ease all;
  :hover {
    border-color: ${props => props.theme.secondary.dark.hex};
    cursor: pointer;
  }
  :focus,
  :checked,
  :active {
    border-color: ${props => props.theme.secondary.dark.hex};
    background-color: ${props => props.theme.secondary.dark.hex};
    + label {
      font-weight: bold;
      color: ${props => props.theme.secondary.dark.hex};
    }
  }
`
StyledRadio.defaultProps = {
  theme: colourPalette.examplePalette,
}
StyledRadio.displayName = "StyledRadio"

// Declared Radio React Component
const Radio = props => {
  // Refactor values out of props (ES6)
  const { className, id, name, onChange, checked, required } = props
  return (
    <StyledRadio
      className={className}
      id={id}
      name={name}
      onChange={onChange}
      checked={checked}
      required={required}
    />
  )
}

// Declare the display name of component for static build
// Display name should match export name
Radio.displayName = "Radio"

// Declare propTypes of Radio
Radio.propTypes = {
  className: PropTypes.string,
  id: PropTypes.string,
  name: PropTypes.string,
  onChange: PropTypes.func,
  checked: PropTypes.bool,
  required: PropTypes.bool,
}

export default Radio
