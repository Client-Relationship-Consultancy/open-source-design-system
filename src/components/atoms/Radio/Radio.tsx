import React from "react";
import styled from "styled-components";

import { colourPalette } from "../../../brandColours";

type InputPropsType = React.HTMLProps<HTMLInputElement>;

const StyledRadio = styled.input.attrs({
  type: "radio",
})`
  appearance: none;
  box-shadow: none;
  border: 2px solid ${props => props.theme.secondary.main.hex};
  outline: none;
  width: 1.5rem;
  height: 1.5rem;
  border-radius: 50%;
  transition: 0.3s ease all;
  :hover {
    border-color: ${props => props.theme.secondary.dark.hex};
    cursor: pointer;
  }
  :focus,
  :checked,
  :active {
    border-color: ${props => props.theme.secondary.dark.hex};
    background-color: ${props => props.theme.secondary.dark.hex};
    + label {
      font-weight: bold;
      color: ${props => props.theme.secondary.dark.hex};
    }
  }
`;
StyledRadio.defaultProps = {
  theme: colourPalette.examplePalette,
};
StyledRadio.displayName = "StyledRadio";

const Radio: React.FunctionComponent<InputPropsType> = props => {
  const { className, id, name, onChange, checked, required } = props;
  return (
    <StyledRadio
      className={className}
      id={id}
      name={name}
      onChange={onChange}
      checked={checked}
      required={required}
    />
  );
};

Radio.displayName = "Radio";

export default Radio;
