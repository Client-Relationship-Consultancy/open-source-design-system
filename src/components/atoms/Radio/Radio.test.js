import React from "react"
import { shallow } from "enzyme"
import "jest-styled-components"
import Radio from "./Radio"

describe("Radio Input test", () => {
  const component = shallow(<Radio name="dog" id="dog" />)
  it("check radio props", () => {
    expect(component.prop("name")).toEqual("dog")
    expect(component.prop("id")).toEqual("dog")
  })
  it("match last snapshot", () => {
    expect(component).toMatchSnapshot()
  })
})
