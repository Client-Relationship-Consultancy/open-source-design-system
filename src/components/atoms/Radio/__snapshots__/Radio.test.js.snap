// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio Input test match last snapshot 1`] = `
<StyledRadio
  id="dog"
  name="dog"
  theme={
    Object {
      "action": Object {
        "dark": Object {
          "RGB": "0,47,108",
          "hex": "#002F6C",
          "on": "#FFFFFF",
        },
        "light": Object {
          "RGB": "98,194,248",
          "hex": "#62C2F8",
          "on": "#FFFFFF",
        },
        "main": Object {
          "RGB": "1,87,155",
          "hex": "#01579B",
          "on": "#FFFFFF",
        },
      },
      "background": Object {
        "RGB": "246,246,246",
        "hex": "#F6F6F6",
      },
      "black": Object {
        "main": Object {
          "RGB": "39,39,39",
          "hex": "#272727",
          "on": "#FFFFFF",
        },
        "tint20": Object {
          "RGB": "236,236,236",
          "hex": "#ECECEC",
          "on": "#272727",
        },
        "tint40": Object {
          "RGB": "199,199,199",
          "hex": "#C7C7C7",
          "on": "#272727",
        },
        "tint60": Object {
          "RGB": "160,160,160",
          "hex": "#A0A0A0",
          "on": "#FFFFFF",
        },
        "tint80": Object {
          "RGB": "105,105,105",
          "hex": "#696969",
          "on": "#FFFFFF",
        },
      },
      "complimentary": Object {
        "dark": Object {
          "RGB": "250,175,64",
          "hex": "#FAAF40",
          "on": "#272727",
        },
        "light": Object {
          "RGB": "249,240,21",
          "hex": "#F9F015",
          "on": "#272727",
        },
        "main": Object {
          "RGB": "253,216,53",
          "hex": "#FDD835",
          "on": "#272727",
        },
      },
      "error": Object {
        "dark": Object {
          "RGB": "143,6,6",
          "hex": "#8F0606",
          "on": "#FFFFFF",
        },
        "main": Object {
          "RGB": "220,16,16",
          "hex": "#DC1010",
          "on": "#FFFFFF",
        },
      },
      "primary": Object {
        "dark": Object {
          "RGB": "11,102,54",
          "hex": "#0B6636",
          "on": "#FFFFFF",
        },
        "light": Object {
          "RGB": "145,208,62",
          "hex": "#91D03E",
          "on": "#FFFFFF",
        },
        "main": Object {
          "RGB": "1,145,68",
          "hex": "#019144",
          "on": "#FFFFFF",
        },
      },
      "secondary": Object {
        "dark": Object {
          "RGB": "72,140,155",
          "hex": "#488C9B",
          "on": "#FFFFFF",
        },
        "light": Object {
          "RGB": "162,235,252",
          "hex": "#A2EBFC",
          "on": "#272727",
        },
        "main": Object {
          "RGB": "118,182,196",
          "hex": "#76B6C4",
          "on": "#FFFFFF",
        },
      },
      "surface": Object {
        "RGB": "255,255,255",
        "hex": "#FFFFFF",
        "on": "#272727",
      },
      "white": Object {
        "RGB": "255,255,255",
        "hex": "#FFFFFF",
        "on": "#272727",
      },
    }
  }
/>
`;
